charts/external-dns/templates/deployment.yaml

```diff
@@ -103,7 +103,7 @@ spec:
             {{- if .Values.txtPrefix }}
             - --txt-prefix={{ .Values.txtPrefix }}
             {{- end }}
-            {{- if and (eq .Values.txtPrefix "") (ne .Values.txtSuffix "") }}
+            {{- if and (not (empty .Values.txtSuffix)) (empty .Values.txtPrefix) }}
```

Comment:
I think for the sake of simplicity this should be linked together.

```helm
{{- if and .Values.txtPrefix .Values.txtSuffix }}
{{- fail (printf "'txtPrefix' and 'txtSuffix' are mutually exclusive") }}
{{- end }}
{{- if .Values.txtPrefix }}
- --txt-prefix={{ .Values.txtPrefix }}
{{- else if .Values.txtSuffix }}
- --txt-suffix={{ .Values.txtSuffix }}
{{- end}}
```

Reply:
Make sense. Refactored and added test